baselines/pFedHN/README.md

```diff
@@ -0,0 +1,96 @@
+---
+title: Personalized Federated Learning using Hypernetworks
+url: https://arxiv.org/abs/2103.04628
+labels: ["data heterogenity", "hypernetworks","personalised federated learning",] # please add between 4 and 10 single-word (maybe two-words) labels (e.g. "system heterogeneity", "image classification", "asynchronous", "weight sharing", "cross-silo")
+dataset: MNIST, CIFAR10, CIFAR100 # list of datasets you include in your baseline
+---
+
+# *pFedHN*
+
+
+
+****Paper:**** https://arxiv.org/abs/2103.04628
+
+****Authors:**** Aviv Shamsian,Aviv Navon,Ethan Fetaya,Gal Chechik
+
+****Abstract:**** Personalized federated learning is tasked with training machine learning models for multiple clients, each with its own data distribution. The goal is to train personalized models in a collaborative way while accounting for data disparities across clients and reducing communication costs. We propose a novel approach to this problem using hypernetworks, termed pFedHN for personalized Federated HyperNetworks. In this approach, a central hypernetwork model is trained to generate a set of models, one model for each client. This architecture provides effective parameter sharing across clients, while maintaining the capacity to generate unique and diverse personal models. Furthermore, since hypernetwork parameters are never transmitted, this approach decouples the communication cost from the trainable model size. We test pFedHN empirically in several personalized federated learning challenges and find that it outperforms previous methods. Finally, since hypernetworks share information across clients we show that pFedHN can generalize better to new clients whose distributions differ from any client observed during training.
+
+## About this baseline
+
+****What’s implemented:**** "The code in the repository reproduces the paper by implementing the concept of hypernetworks which create weights for that target network thus resolving the problems of dataheterogenity.The hypernetworks lies in the server and the clients have the target model. Initially hypernetwork sends the weights from the server which is loaded into the target net . After the targetnet is trained we pass the delta_theta to the client , which inturn updated the phi_gradients for the hypernetwork and the flow goes on."
+
+****Datasets:**** MNIST,CIFAR10,CIFAR100 from torchvision 
+
+****Hardware Setup:**** Will be updated
```

Comment:
👍 , please indicate if GPU is needed to run these experiments and also what's the expected duration to finish the experiments.